docs/src/main/tut/dsl.md

```diff
@@ -392,4 +392,48 @@ val averageTemperatureService = HttpService[IO] {
 }
 ```
 
+#### Optional query parameters
+
+To accept a optional query parameter a `OptionalQueryParamDecoderMatcher` can be used.
+
+```tut:book
+object OptionalYearQueryParamMatcher extends OptionalQueryParamDecoderMatcher[Year]("name")
+
+def getAverageTemperatureForCurrentYear(): IO[String] = ???
+def getAverageTemperatureForYear(y: Year): IO[String] = ???
+
+val service = HttpRoutes.of[IO] {
+  case GET -> Root / "temperature" :? OptionalYearQueryParamMatcher(maybeYear) =>
+    maybeYear match {
+      case None =>
+        Ok(getAverageTemperatureForCurrentYear())
+      case Some(year) =>
+        Ok(getAverageTemperatureForYear(year))
+    }
+}
+```
+
+#### Missing required query parameters
+
+A request with a missing required query parameter will be fall through to the following `case` statements and may eventually return a 404. To provide contextual error handling, optional query parameters or fallback routes can be used.
+
+#### Invalid query parameter handling
+
+To validate query parsing you can use `ValidatingQueryParamDecoderMatcher` which returns a `ParseFailure` if the parameter cannot be decoded.
```

Comment:
It may be worth mentioning that the mapping to `BadRequest` shouldn't echo any values from the request for [XSS  reasons](https://www.owasp.org/index.php/Cross-site_Scripting_(XSS)).